web3/eth.py

```diff
@@ -225,9 +225,11 @@ def getTransactionByBlock(self, block_identifier, transaction_index):
             [block_identifier, transaction_index],
         )
 
-    def waitForTransactionReceipt(self, transaction_hash, timeout=120):
+    def waitForTransactionReceipt(self, transaction_hash, timeout=120, poll_latency=0.2):
         try:
-            return wait_for_transaction_receipt(self.web3, transaction_hash, timeout)
+            if poll_latency > timeout:
+                poll_latency = timeout
```

Comment:
Good catch. I think this logic could be moved to `wait_for_transaction_receipt`.